docs/architecture/layout.md

```diff
@@ -0,0 +1,59 @@
+# Overview
+
+This is a basic layout for NGINX Agent application. This document focusing only on the general layout and not what you have inside. It`s basic so that the information is quickly digestable. 
+
+## Directories
+
+### `/cmd`
+
+Main applications for this project.
+
+The directory name for each application should match the name of the executable (e.g., /cmd/agent).
+
+### `/internal`
+
+Private application and library code. This is the code will not be exported and available outside of this project.
+
+### `/pkg`
+
+Library code to be used by external applications (e.g., /pkg/files and /pkg/uuid). Other projects will import these libraries.
+
+This is a way of grouping public facing Go code in one place away from the root directory. The root directory may contain non-Go components and directories making it easier to run various Go tools (as mentioned in these talks: Best Practices for Industrial Programming from GopherCon EU 2018, GopherCon 2018: Kat Zien - How Do You Structure Your Go Apps and GoLab 2018 - Massimiliano Pippi - Project layout patterns in Go).
+
+The pkg directory origins: The old Go source code used to use pkg for its packages and then various Go projects in the community started copying the pattern (see this Brad Fitzpatrick`s tweet for more context).
+
+### `/vendor`
```

Comment:
Is this needed as we no longer have a vendor folder ?

Reply:
Are we meant to have one locally ? I thought it wasn't supposed to be there at all, not just that it wasn't checked in

Reply:
there should be a local vendor when you do `go mod vendor` locally.

Reply:
You don't need to run `go mod vendor` locally though.

Reply:
it's still worth keeping it in the notes? Just in case some Go developers are wondering